docs/tools/terraform/v1v2.rst

```diff
@@ -0,0 +1,216 @@
+
+We can remove the old Terraform folder ``rm -rf ~/.terraform.d``
+
+Edit the providers block of your script to include the latest version of
+the Aiven Terraform Provider (v2.3.1 as of 5th November 2021)
+
+::
+
+    terraform {
+      required_providers {
+        aiven = {
+          source = "aiven/aiven"
+          version = "2.3.1"
+        }
+      }
+    }
+
+After that you can run: ``terraform 0.13upgrade`` to see fixes
+recommended by Hashicorp. More information
+`here <https://www.terraform.io/upgrade-guides/0-13.html>`__.
+
+Bump it to the latest
+~~~~~~~~~~~~~~~~~~~~~
+
+Any version above 0.13 will be fine, here we use the latest (1.0.10 as
+of 5th November 2021) ``tfenv install latest && tfenv use latest``
+
+Run ``terraform init -upgrade``
+
+Run ``terraform plan``
+
+.. image:: /images/tools/terraform/terraform-upgrade.jpg
+   :alt:
+
+You may see warnings or errors like the above, these will point towards
+changes made between the release you are running and the latest release.
+
+The warnings will provide recommendations on the changes to make and you
+can get more information using our
+`docs <https://registry.terraform.io/providers/aiven/aiven/latest/docs>`__.
+
+Migrating away from ``aiven_service`` to the ``aiven_X`` service specific resource
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Migration strategy – since ``aiven_service`` and ``aiven_X``, i.e.
+``aiven_kafka`` are different kind of resources just by rewriting code
+we would cause destructive actions.
+
+Also running ``terraform state mv <a> <b>`` will not do it because it is
+a different resource type.
+
+To solve it we would have to:
+
+-  Change the code first
+-  Remove old resource from the state
+-  Import already existing service to the Terraform state.
+
+To change from the old ``aiven_service.kf`` to the new ``aiven_kafka``
+resource, resource type should be changed as well as ``service_type``
+should be removed.
+
+Also all the references should be updated from the
+``aiven_service.kf.x`` to the ``aiven_kafka.kf.x``.
+
+::
+
+    - resource "aiven_service" "kafka" {
+    -    service_type            = "kafka"
+    + resource "aiven_kafka" "kafka" {
+        ...
+    }
+    resource "aiven_service_user" "kafka_user" {
+      project      = var.aiven_project_name
+    -  service_name = aiven_service.kafka.service_name
+    +  service_name = aiven_kafka.kafka.service_name
+      username     = var.kafka_user_name
+    }
+
+Here is the example of the steps above:
+
+::
+
+    $ terraform state list | grep kf
+    $ terraform state rm aiven_service.kf
+
+    $ terraform import aiven_kafka.kf mischa-demo/kafka-test-gr0gu
+    $ terraform plan
+    $ terraform apply
+
+For more examples please check Aiven Examples
+`repository <https://github.com/aiven/aiven-examples/tree/master/terraform>`__
+on GitHub.
+
+If you have any issues, please contact our `support <mailto:support@aiven.io>`__.
```

Comment:
The last line isn't needed, let's stick specifically to the steps needed for the provider upgrade.